Description: use busybox ash's echo for awk.tests

dash's echo doesn't understand the -e flag, so this test suite uses a simple
self-built echo binary (scripts/echo.c) instead. However that version cannot
output NULs with '\0' which breaks the "awk printf('%c') can output NUL" test.
Force this file to use busybox ash and re-detect what to do with echo to avoid
this issue.

Author: Chris Boot <bootc@debian.org>
Forwarded: http://lists.busybox.net/pipermail/busybox/2021-December/089391.html
Last-Update: 2021-12-27

--- busybox-1.35.0.orig/testsuite/awk.tests
+++ busybox-1.35.0/testsuite/awk.tests
@@ -1,4 +1,10 @@
-#!/bin/sh
+#!/usr/bin/env ash
+unset ECHO
+# Debian hack: dash's echo doesn't understand the -e flag, so this test suite
+# uses a simple self-built echo binary (scripts/echo.c) instead. However that
+# version cannot output NULs with '\0' which breaks the "awk printf('%c') can
+# output NUL" test. Force this file to use busybox ash and re-detect what to do
+# with echo to avoid this issue.
 
 # Copyright 2007 by Denys Vlasenko <vda.linux@googlemail.com>
 # Licensed under GPLv2, see file LICENSE in this source tree.
